consensus/src/lib.rs

```diff
@@ -102,17 +103,20 @@ pub enum FinalizeBlockInputData {
     None,
 }
 
-pub fn generate_consensus_data_and_reward<G>(
+#[allow(clippy::too_many_arguments)]
+pub fn generate_consensus_data_and_reward<G, R>(
     chain_config: &ChainConfig,
     prev_block_index: &GenBlockIndex,
     sealed_epoch_randomness: PoSRandomness,
     input_data: GenerateBlockInputData,
     block_timestamp: BlockTimestamp,
     block_height: BlockHeight,
     get_ancestor: G,
+    rng: R,
```

Comment:
it looks like this function can create true rng inside, same as `finalize_consensus_data`

Reply:
Maybe. I don't think it matters.

Reply:
just consistency, otherwise look suspicious